docs/reference/filebeat/file-identity.md

```diff
@@ -0,0 +1,183 @@
+
+This can be fixed by using a special marker file on your file system instead of asking the OS for a device number.
+
+Users specify a path by setting `file_identity.inode_marker.path`. This path leads to a file that has some short unique text that remains unchanged forever. The content of this file replaces the `device` number in the `native` file identity. The rest remains the same as `native`.
+
+### `fingerprint`
+
+The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.
+
+This file identity is using file fingerprints produced by the scanner component of the filestream input. Therefore, in order to activate this file identity, it's required to first activate fingerprint in the scanner by setting `prospector.scanner.fingerprint.enabled: true`.
+
+In the context of the filestream input, fingerprint is a hash string computed from a part of the file content.
+
+![Fingerprint File Identity](/reference/filebeat/img/fingerprint-file-identity.png)
+
+By default, the hashed part of the file is the first 1024 bytes (`offset: 0`, `length: 1024`). Most of the log files have timestamps and other unique segments on each line that make sure that the first 1024 bytes combined together constitute a unique set of characters. This makes sure that a produced fingerprint is a stable unique identifier for each file.
+
+If necessary it's possible to move this hashed byte range farther in the file by adjusting `prospector.scanner.fingerprint.offset`. It's also possible to change the amount of bytes used for hashing by adjusting `prospector.scanner.fingerprint.length` (minimum value is 64). Keep in mind, the smaller the range, the more likely it is to hit a collision, so the fingerprint (ID) is not unique anymore. If this happens only the first file with a collided ID will be ingested.
+
+This file identity covers most of the use-cases. It's OS-agnostic, supports any file system including network shares and it makes sure to avoid data duplication or data loss.
+
+However, there is a trade off: using this file identity delays ingesting data from files smaller than the required amount of bytes for computing the fingerprint.
+
+For example, Filebeat is running a filestream input with this configuration:
+
+```yaml
+prospector.scanner.fingerprint:
+  enabled: true
+  offset: 32   # e.g. the first 32 bytes are always identical in every file, we skip them
+  length: 1024 # strong guarantee of uniqueness
+file_identity.fingerprint: ~
+```
+
+If a new log file appears, the filestream input would not "see" it until it grows to 1056 bytes (`offset`+`length`).
+
+For some use-cases it might be a requirement to receive data as soon as possible without delays. For some use-cases files would not grow to such sizes at all and would never be ingested. It's important to consider these limitations accordingly.
+
+The stability of this file identity makes it a recommended and default option.
```

Comment:
The whole document is great @rdner !

Aside from the minor changes I pointed out, I'd also move fingerprint to the top, so users are first introduced to our recommend, file identity, then others can be discussed. I'd also move this phrase to the top, stating on the same paragraph that this  is the default for 9.0 and is the recommended one.

Reply:
I meant to order them in ascending complexity and reliability. I also believe it's a chronological order of when they were introduced.

I put the overview table first to have better visibility, I can mark defaults with bold.

Reply:
Done in https://github.com/elastic/beats/pull/43609/commits/32c62fcd11d792a9cc449282160f4394bf714e66

Reply:
This guide is supposed to help choose the right tool for the right use-case not to push customers to using the only option we want them to use. As this guide explains `fingerprint` might be not the right option for some use-cases (e.g. small files and instant ingest).